### PR TITLE
Add repro to undo move dashboard doesn't always work in v44

### LINKED ISSUE #25705: Undo move dashboard doesn't always work in v44
**Describe the bug**
When you move a dashboard there is a toast popup in the lower left corner of the window saying to which collection was dashboard moved. That popup also provides the option to "Undo" that action. It seems that this functionality has been broken in the v44.

**Logs**
The `PUT` request when you hit the "Undo" button is empty `{}`

**To Reproduce**
1. Have a dashboard with a question in "Our analytics" and have at least one more collection (let's call it "Other collection")
2. Open a dashboard by pasting a link to it rather then clicking on it from the root collection (not sure why but this step is crucial and is the only way to reproduce this issue)
3. Click on the `...` menu and choose "Move"
4. Choose "Other collection" and then click "Move"
5. In the lower left corner, click "Undo"
6. Note that the breadcrumb still says "Other collection" instead of "Our analytics" and note that the dashboard actually didn't move

**Expected behavior**
Undo button should move the dashboard back to the original collection.

**Information about your Metabase Installation:**
local dev, `release-x.44.x` branch, bdd15015054cd75736dba7de67343efbbbb1d90d, H2, Sample Database

**Severity**
P3

**Additional context**
- Doesn't seem to happen on `master` (v45)`
- This was caught by Cypress because the test for moving and undo moving the dashboard is failing consistently. At first I thought it's some weird race condition but then I could reproduce it manually.

### PR DESCRIPTION
**Pull request by WiNloSt**
Closes https://github.com/metabase/metabase/issues/25705

### Description

Add tests to https://github.com/metabase/metabase/issues/25705.

This doesn't seem to happen in `master` @ 8d366f210c04d847bb5011c249e10b9b64510421 nor 0.46 (don't remember the version, but I downloaded it from [this page](https://www.metabase.com/start/oss/))

Can reproduce with this jar [v0.44.7.3](https://github.com/metabase/metabase/releases/tag/v0.44.7.3)

### How to verify

1. Run test from this PR

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/support/helpers/e2e-dashboard-helpers.js [hunk 1]
```diff
@@ -175,3 +175,7 @@ export function resizeDashboardCard({ card, x, y }) {
 export function createLinkCard() {
   cy.icon("link").click();
 }
+
+export function openDashboardMenu() {
+  return dashboardHeader().findByLabelText("dashboard-menu-button");
```
Comment (WiNloSt) [inline e2e/support/helpers/e2e-dashboard-helpers.js:180]: There are some usage like this

https://github.com/metabase/metabase/blob/c9134399b7fb0ced0d667a20e558f8ada926f6a7/e2e/test/scenarios/sharing/downloads/downloads.cy.spec.js#L180

**Comment by uladzimirdev** [comment]
@WiNloSt does the test fail on 0.44?

**Comment by WiNloSt** [comment]
@uladzimirdev 
> @WiNloSt does the test fail on 0.44?

I've tried running that on 0.44 with no success. We have Cypress in a different directory then. I couldn't even run `yarn test-cypress-open` on 44 so I gave up.

**Comment by WiNloSt** [comment]
@uladzimirdev After running `bin/build-drivers.sh` I'm able to run Cypress in 44. Even though, some E2E helpers are missing, I added them manually but the test body stays intact, and the test failed as expected.
Screenshot 2023-08-03 at 2 55 50 PM

File: e2e/support/helpers/e2e-dashboard-helpers.js [hunk 2]
```diff
@@ -183,3 +183,7 @@ export function resizeDashboardCard({ card, x, y }) {
 export function createLinkCard() {
   cy.icon("link").click();
 }
+
+export function openDashboardMenu() {
+  return dashboardHeader().findByLabelText("dashboard-menu-button").click();
```
Comment (WiNloSt) [inline e2e/support/helpers/e2e-dashboard-helpers.js:188]: There are some usage like this

https://github.com/metabase/metabase/blob/c9134399b7fb0ced0d667a20e558f8ada926f6a7/e2e/test/scenarios/sharing/downloads/downloads.cy.spec.js#L180

File: e2e/test/scenarios/dashboard/reproductions/25705-undo-moving-dashboards.cy.spec.js
Comment (WiNloSt) [inline e2e/test/scenarios/dashboard/reproductions/25705-undo-moving-dashboards.cy.spec.js:1]: This test failed successfully on 44. Please see https://github.com/metabase/metabase/pull/32835#issuecomment-1663473106
image

**Review by ranquild — APPROVED** [review]
Maybe add `cy.wait` for update requests?

**Comment by WiNloSt** [comment]
@ranquild
> Maybe add `cy.wait` for update requests?

 `*.findBy***` should work for that. There's a comment [I made replying to Ryan's Slack message.](https://metaboat.slack.com/archives/C5XHN8GLW/p1690427262608129?thread_ts=1690409097.253249&cid=C5XHN8GLW). I haven't replied to his latest comment yet, as I want to figure out how `findBy***` works in different scenarios.
 
But from my experience, `*.findBy***` generally retries to find the element until it reaches the timeout.

The only practical difference I can see at the moment is the [default timeout](https://docs.cypress.io/guides/references/configuration#Timeouts).
- DOM commands have a default timeout `defaultCommandTimeout` of `4000`ms
- `cy.wait` has a default timeout `requestTimeout` of `5000`ms

I find using `*.findBy***` more reliable in most cases since we don't need to know the underlying API. However, I understand there might be cases which `cy.wait` is required, but it should not be necessary in this case.

**Comment by uladzimirdev** [comment]
> @uladzimirdev After running `bin/build-drivers.sh` I'm able to run Cypress in 44. Even though some E2E helpers are missing, I added them manually but the test body stays intact, and the test failed as expected.
> 
> Screenshot 2023-08-04 at 2 41 22 PM

@WiNloSt btw, have you tried using proxy and metabase jar? 😄

**Pull request merged**